src/Stormpath/DataStore/PropertyBasedClassNameResolver.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/*
+ * Copyright 2013 Stormpath, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Stormpath\DataStore;
+
+
+/**
+ * Helper for the <code>DataStore</code> to resolve the class name of a resource
+ * to instantiate based on the value of a property in the data set received.
+ *
+ * @package Stormpath\DataStore
+ */
+interface PropertyBasedClassNameResolver
```

Comment:
This is really a generic class name resolver, so I would rename it to `ClassNameResolver`. Also remember to rename the default implementation and the documentation of the class.